src/Imgix/UrlBuilder.cs

```diff
@@ -56,40 +58,217 @@ public String BuildUrl(String path)
             return BuildUrl(path, new Dictionary<string, string>());
         }
 
+        /// <summary>
+        /// Create a resolvable imgix url.
+        /// 
+        /// </summary>
+        /// <param name="path">path to the image, i.e. "image/file.png"</param>
+        /// <param name="parameters">dictionary of query parameters</param>
+        /// <returns>resolvable URL</returns>
+
+        /*
+        * Encodes the `path` and `paramater` arguments to the UTF8 
+        * scheme and returns a formatted string that conforms to the 
+        * "{scheme}://{domain}/{path}{params}" format.
+        */
         public String BuildUrl(String path, Dictionary<String, String> parameters)
         {
+            String scheme = UseHttps ? "https" : "http";
+            path = SanitizePath(path);
+
+            var qs = BuildParams(parameters);
+            var localParams = new Dictionary<String, String>(parameters);
 
-            if (path.StartsWith("http"))
+            if (!String.IsNullOrEmpty(_signKey))
             {
-                path = WebUtility.UrlEncode(path);
+                var hashString = String.Format("{0}/{1}{2}", _signKey, path, localParams.Any() ? "?" + qs : String.Empty);
+                localParams.Add("s", HashString(hashString));
             }
+            return String.Format("{0}://{1}/{2}{3}", scheme, Domain, path, localParams.Any() ? "?" + BuildParams(localParams) : String.Empty);
```

Comment:
Same comment as above.

```suggestion
            var queryString = localParams.Any() ? "?" + BuildParams(localParams) : String.Empty;
            return String.Format("{0}://{1}/{2}{3}", scheme, Domain, path, queryString);
```